### PR TITLE
Modules/data analytics SRS introduction

### PR DESCRIPTION
**Pull request by GaganSinghGit**
# Description

Data analytics SRS introduction

Fixes # (issue)

## Type of change

- [ ] Documentation (update or new)

### CODE REVIEW

**Comment by Zephreo** [comment]
You need to run prettier on the file:
- Open the documentation folder in your terminal (that has npm installed)
- `npm install`
- `npm run format`
- `npm run lint`
- Commit changes

**Review by Zephreo — CHANGES_REQUESTED** [review]
You need to run prettier on the file:
- Open the documentation folder in your terminal (that has npm installed)
- `npm install`
- `npm run format`
- `npm run lint`
- Commit changes

Otherwise drop me or Tim a message if you need help setting that up

File: docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md [hunk 1]
```diff
@@ -2,16 +2,42 @@
...
 
+- Create dataframes by importing datasets. Adding and removing rows and columns on the dataset.
+- Overwriting data inside the dataset.
+- Preprocessing functionality for cleaning and standardizing the dataframe for the goal of making it
+  fit to be used in a machine learning model.
+- Create data visualisations like bar graphs, histograms.
+
 ### 1.5 Definitions and Acronyms
 
+- Dataset: A collection of data in tabular form. fFiles ending in comma separated values (.csv), tab
```
Comment (timmydzeko) [inline docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md:34]: Need to remove lowercase f from beginning of `Files` on line 34

File: docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md [hunk 2]
```diff
@@ -2,16 +2,42 @@
...
+  separated values (.tbv), Excel (.xlsx) file extension.
```
Comment (timmydzeko) [inline docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md:35]: Add .txt files as an option as well

**Review by timmydzeko — CHANGES_REQUESTED** [review]

File: docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md [hunk 3]
```diff
@@ -2,16 +2,42 @@
...
 
+The purpose of Data Analytics module is to allowing users to load datasets from various file
+formats, then perform data manipulations and visualization for the goal of extracting useful
+insights from the dataset.
+
 ### 1.1 Purpose
 
+The purpose of Data Analytics module is that it is a tools that allows users to import crate data
+visualisations and use pre written algorithms to perform statistical analysis of the data, and
+pre-process the data for use in machine learning model.
```
Comment (Zephreo) [inline docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md:13]:
```suggestion
The purpose of Data Analytics module is that it is a tool that allows users to import, create data
visualisations, and use pre-written algorithms to perform statistical analysis of the data, and
pre-process data for use in a machine learning model.
```

File: docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md [hunk 4]
```diff
@@ -2,16 +2,42 @@
...
+
 ### 1.2 Intended Audience
 
+The intended audience is Data Analysts, Machine Learning Engineer and developer.
```
Comment (Zephreo) [inline docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md:17]:
```suggestion
The intended audience is Data Analysts, Machine Learning Engineers, and developers.
```

File: docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md [hunk 5]
```diff
@@ -2,16 +2,42 @@
...
+
 ### 1.3 Intended Use
 
+The intended use of this module is to create data visualisations and prepossess the data to be used
+in a machine learning model.
```
Comment (Zephreo) [inline docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md:22]:
```suggestion
The intended use of this module is to create data visualisations and preprocess the data to be used
in a machine learning model.
```
Needs check from Tim about ML models here.

File: docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md [hunk 6]
```diff
@@ -2,16 +2,42 @@
...
+
 ### 1.4 Scope
 
+- Create dataframes by importing datasets. Adding and removing rows and columns on the dataset.
+- Overwriting data inside the dataset.
+- Preprocessing functionality for cleaning and standardizing the dataframe for the goal of making it
+  fit to be used in a machine learning model.
+- Create data visualisations like bar graphs, histograms.
```
Comment (Zephreo) [inline docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md:30]:
```suggestion
- Create dataframes by importing datasets. Adding and removing rows/columns in a dataset.
- Changing data inside a dataset.
- Preprocessing functionality for cleaning and standardising the dataframe for the goal of making it
  fit to be used in a machine learning model.
- Create data visualisations like bar graphs, histograms.
```

File: docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md [hunk 7]
```diff
@@ -2,16 +2,42 @@
...
+
 ### 1.5 Definitions and Acronyms
 
+- Dataset: A collection of data in tabular form. Files ending in comma separated values (.csv), tab
+  separated values (.tbv), Excel (.xlsx), test (.txt) file extension.
+- Dataframe: Representation of dataset as an object in the program which allows users to use method
+  calls to preform actions on the dataset.
+- Standardizing: the process of converting data to a common format to enable users to process and
+  analyse it
```
Comment (Zephreo) [inline docs/Splashkit/Modules/Data Analytics/Software_Requirements_Specification.md:39]:
```suggestion
- Dataset: A collection of data in tabular form. Files ending in comma separated values (.csv), tab
  separated values (.tbv), Excel (.xlsx), test (.txt) file extension.
- Dataframe: Representation of dataset as an object in the program which allows users to use method
  calls to perform actions on the dataset.
- Standardising: the process of converting data to a common format to enable users to process and
  analyse it
```

**Review by Zephreo — CHANGES_REQUESTED** [review]

**Review by Zephreo — APPROVED** [review]
I don't see any glaring issues, am happy for merge after tim checks.

**Review by timmydzeko — APPROVED** [review]

**Comment by timmydzeko** [comment]
@GaganSinghGit The conflict is that the file location has been changed. Needs to be committed into a folder called Background inside the Data Analytics folder. This should fix the merge conflict

**Comment by Zephreo** [comment]
just resolved conflict, sory

**Pull request merged**